eo-runtime/src/main/java/org/eolang/PhSafe.java

```diff
@@ -197,25 +199,38 @@ private <T> T through(final Action<T> action, final String suffix) {
                 this.label(suffix)
             );
         } catch (final Throwable ex) {
-            final StringBuilder msg = new StringBuilder(0);
-            final StackTraceElement[] stack = ex.getStackTrace();
-            if (stack != null && stack.length > 0) {
-                final StackTraceElement last = stack[0];
-                msg.append(last.getFileName())
-                    .append(':')
-                    .append(last.getLineNumber())
-                    .append(": ");
-            }
-            msg.append(ex.getClass().getSimpleName())
-                .append(": ")
-                .append(ex.getMessage());
             throw new EOerror.ExError(
-                new Data.ToPhi(msg.toString()),
-                this.label(suffix)
+                new Data.ToPhi(ex.getMessage()),
+                trace(ex, this.label(suffix))
```

Comment:
@yegor256 let's make it `PhSafe.trace()`

Reply:
@maxonfjvipon fixed in 698c11f9d